cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1328,20 +1367,17 @@ func mountNonDataPartitionMatchingKernelDisk(dir, fallbacklabel string) error {
 	//       to use by-uuid or by-id
 	partSrc := filepath.Join("/dev/disk/by-partuuid", partuuid)
 	if err != nil {
-		// no luck, try mounting by label instead
-		partSrc = filepath.Join("/dev/disk/by-label", fallbacklabel)
+		partSrc, err = getNonUEFISystemDisk(fallbacklabel)
+		if err != nil {
+			return err
+		}
```

Comment:
nitpick: this code was already a bit unusual because of the var setting in-between the error checking. a slightly longer more expected code would:

```
        var partSrc string
        partuuid, err := bootFindPartitionUUIDForBootedKernelDisk()
        // TODO: the by-partuuid is only available on gpt disks, on mbr we need
        //       to use by-uuid or by-id
        if err == nil  {
                partSrc = filepath.Join("/dev/disk/by-partuuid", partuuid)
       } else {
                partSrc, err = getNonUEFISystemDisk(fallbacklabel)
                if err != nil {
                        return err
                }
        }

```